src/i965_encoder_common.h

```diff
@@ -0,0 +1,517 @@
+
+    void (*pfn_set_curbe_brc_frame_update)(
+        VADriverContextP ctx,
+        struct encode_state *encode_state,
+        struct i965_gpe_context *gpe_context,
+        struct intel_encoder_context *encoder_context,
+        void *param);
+
+    void (*pfn_set_curbe_brc_mb_update)(
+        VADriverContextP ctx,
+        struct encode_state *encode_state,
+        struct i965_gpe_context *gpe_context,
+        struct intel_encoder_context *encoder_context,
+        void *param);
+
+    void (*pfn_set_curbe_sfd)(
+        VADriverContextP ctx,
+        struct encode_state *encode_state,
+        struct i965_gpe_context *gpe_context,
+        struct intel_encoder_context *encoder_context,
+        void *param);
+
+    void (*pfn_set_curbe_wp)(
+        VADriverContextP ctx,
+        struct encode_state *encode_state,
+        struct i965_gpe_context *gpe_context,
+        struct intel_encoder_context *encoder_context,
+        void *param);
+
+    //surface set function pointer
+    void (*pfn_send_scaling_surface)(
+        VADriverContextP ctx,
+        struct encode_state *encode_state,
+        struct i965_gpe_context *gpe_context,
+        struct intel_encoder_context *encoder_context,
+        void *param);
+
+    void (*pfn_send_me_surface)(
+        VADriverContextP ctx,
+        struct encode_state *encode_state,
+        struct i965_gpe_context *gpe_context,
+        struct intel_encoder_context *encoder_context,
+        void *param);
+
+    void (*pfn_send_mbenc_surface)(
+        VADriverContextP ctx,
+        struct encode_state *encode_state,
+        struct i965_gpe_context *gpe_context,
+        struct intel_encoder_context *encoder_context,
+        void *param);
+
+    void (*pfn_send_brc_init_reset_surface)(
+        VADriverContextP ctx,
+        struct encode_state *encode_state,
+        struct i965_gpe_context *gpe_context,
+        struct intel_encoder_context *encoder_context,
+        void *param);
+
+    void (*pfn_send_brc_frame_update_surface)(
+        VADriverContextP ctx,
+        struct encode_state *encode_state,
+        struct i965_gpe_context *gpe_context,
+        struct intel_encoder_context *encoder_context,
+        void *param);
+
+    void (*pfn_send_brc_mb_update_surface)(
+        VADriverContextP ctx,
+        struct encode_state *encode_state,
+        struct i965_gpe_context *gpe_context,
+        struct intel_encoder_context *encoder_context,
+        void *param);
+
+    void (*pfn_send_sfd_surface)(
+        VADriverContextP ctx,
+        struct encode_state *encode_state,
+        struct i965_gpe_context *gpe_context,
+        struct intel_encoder_context *encoder_context,
+        void *param);
+
+    void (*pfn_send_wp_surface)(
+        VADriverContextP ctx,
+        struct encode_state *encode_state,
+        struct i965_gpe_context *gpe_context,
+        struct intel_encoder_context *encoder_context,
+        void *param);
+};
+/*
+   the definition for encoder codec state
+*/
+
```

Comment:
for fame_width_in_pixel and frame_height_in_pixel and etc, these variabls are used almost in each function. it is easy to acces and read to put them together in one structure. I think it is OK.

for frame_width_in_mbs, it is common for AVC/VP8, and even if HEVC/VP9, it is also used on some caculation when MB alignment. 
frame_width_in_ctb/cb/cu/lcu is specified for HEVC only,VP should use SB related. that is the reason there are no definition in this structure generic_enc_codec_state here.